pandas/core/window.py

```diff
@@ -148,16 +148,17 @@ def _shallow_copy(self, obj=None, **kwargs):
                 kwargs[attr] = getattr(self, attr)
         return self._constructor(obj, **kwargs)
 
-    def _prep_values(self, values=None, kill_inf=True, how=None):
+    def _prep_values(self, values=None, kill_inf=True, how=None,
+                     as_float=True):
```

Comment:
remove the ``as_float`` arg (it will simply coerce ATM for integer types, no easy way around this)